### ISSUE TITLE
Controller support for Moonlight using Parsec

### ISSUE THREAD

**Issue by krostitzer** [acceleration3/cloudgamestream#11]
@acceleration3 Thanks for your script. It helped me set up moonlight for my **Google cloud T4 VM** using Parsec. Everything works fine when I connect to the **moonlight** client on my MacOS except game **controller**. I use a wired Xbox controller and it works perfectly fine with GTA V if I play in **Parsec** but as you know already, the video quality gets choppy quite frequently. I am at the end of my wits here, have already tried everything I knew. If you could **help** in any way, it would be really great!!

**Comment by acceleration3** [comment]
This appears to be a bug with GameStream itself. Have you tried the fix that involves renaming the Shield controller DLL files so they don't get used? I'm referring to the one in [this post in the NVIDIA forums](https://www.nvidia.com/en-us/geforce/forums/gamestream/19/279241/can-we-have-an-option-to-disable-the-shield-contro/2040485/).

**Comment by krostitzer** [comment]
Just applied this fix but nothing changed. The controller still works on Parsec but not on Moonlight. I used gamepadtester.com to test the controller.
I am using GFE 3.13.0.85 and NVIDIA Driver 461.40

**Comment by krostitzer** [comment]
So is this a known bug? Were you able to find any other work through around this?

**Comment by acceleration3** [comment]
I need more info on this, did you check if the files are in the folders and are renamed? Does the controller get detected at all before and after the renaming? Are you opening Steam and do you have the controller configuration enabled?

**Comment by krostitzer** [comment]
Yes, I confirm that the files have been renamed in both folders. 
The controller doesn't get recognised at all (neither on gamepadtester.com and nor in the game).
I am not using Steam at all. I run my game through the Epic store.

Thanks for your reply. Please let me know if I missed something.

**Comment by acceleration3** [comment]
I can suggest trying [VirtualHere](https://virtualhere.com/osx_server_software) to share your controller over the internet, although it would require you to forward a port (7575 TCP) on your MacOS system and I've seen that Xbox One wired controllers appear to have some incompatibility issues and can only be used with their dongle.

**Comment by krostitzer** [comment]
Thanks for the suggestions. My controller doesn't require a dongle. Its a bluetooth Xbox One controller but I use with a USB-C cable. How do I forward ports on MacOS

What are your thoughts on increasing the video quality on Parsec instead? If the choppy artefacts could be removed, it works quite well latency wise for me.

**Comment by acceleration3** [comment]
Port forwarding is done by accessing your router's gateway webpage. If by choppy artifacts you mean stuttering, then it's possible the fix is to disable MacOS's location services while you use Parsec and reenabling them when you finish the session.
